pkg/backend/identityapi/register_client.go

```diff
@@ -13,16 +14,33 @@ import (
 	"strings"
 	"time"
 
+	"github.com/antihax/optional"
 	backendhttp "github.com/newrelic/infrastructure-agent/pkg/backend/http"
 	"github.com/newrelic/infrastructure-agent/pkg/backend/inventoryapi"
 	"github.com/newrelic/infrastructure-agent/pkg/entity"
+	"github.com/newrelic/infrastructure-agent/pkg/identity-client"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
 	"github.com/newrelic/infrastructure-agent/pkg/log"
 )
 
-var rlog = log.WithComponent("IdentityRegisterClient")
+var rlog = log.WithComponent("identityapi.RegisterClient")
 
-type IdentityRegisterClient interface {
-	Register(agentEntityID entity.ID, entities []RegisterEntity) ([]RegisterEntityResponse, time.Duration, error)
+const (
+	identityPath = "/identity/v1"
+)
+
+// RegisterClient provides the ability to register either a single entity or a
+// "batch" of entities.
+type RegisterClient interface {
+
+	// Deprecated: method to be removed at the end of this completing this feature
+	RegisterEntitiesRemoveMe(agentEntityID entity.ID, entities []RegisterEntity) ([]RegisterEntityResponse, time.Duration, error)
+
+	// RegisterBatchEntities registers a slice of protocol.Entity. This is done as a batch process
+	RegisterBatchEntities(agentEntityID entity.ID, entities []protocol.Entity) (RegisterBatchEntityResponse, time.Duration, error)
+
+	// RegisterEntity registers a protocol.Entity
+	RegisterEntity(agentEntityID entity.ID, entity protocol.Entity) (RegisterEntityResponse, error)
```

Comment:
I don't see this being used, what's the use case for it?

Reply:
We originally started by registering one (to check it worked) and then moved to batch. Can probably be removed now unless we plan to add ability to register one entity (then again you can just have a slice of one).